### PR TITLE
[Storage] Testing various service versions.

### PR DESCRIPTION
**Pull request by kasobol-msft**
This PR will work with https://github.com/Azure/azure-sdk-for-java/pull/21321 .
In this PR:
- plumb service version into client builder instrumentation if service version has been defined in env otherwise don't (i.e. test latest by default)
- ignore tests that do something with service version explicitly if one is set
- Fix bug in datalake where service version wasn't passed correctly in some cases.
- Fix bug in batch where service version wasn't passed correctly in some cases.
- Fix few tests that didn't instrument correctly.

### CODE REVIEW

**Comment by kasobol-msft** [comment]
java - storage - tests

File: sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncyptedBlockBlobAPITest.groovy [hunk 1]
```diff
@@ -1489,6 +1490,7 @@ class EncyptedBlockBlobAPITest extends APISpec {
         }
     }
 
+    @IgnoreIf( { getEnv().serviceVersion != null } )
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncyptedBlockBlobAPITest.groovy:1493]: Does the `getEnv().serviceVersion` not get overridden by the testing policy?

**Review by alzimmermsft — APPROVED** [review]

File: sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncyptedBlockBlobAPITest.groovy (hunk 1 above)
Comment (kasobol-msft) [reply sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncyptedBlockBlobAPITest.groovy:1493]: It's either null or something depending on env.
https://github.com/kasobol-msft/azure-sdk-for-java/blob/2eb4f25a831128d403913179cd8ff33b4c8b44c9/sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/TestEnvironment.java#L68-L77

**Comment by kasobol-msft** [comment]
image

**Pull request merged**